### PR TITLE
[Docs] Add phpdoc

### PR DESCRIPTION
**Pull request by chtipepere**
First PR to get in touch.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@ composer.lock
 Guardfile
 phpunit.xml
 /vendor/
+.idea/
```
Comment (pablodip) [inline .gitignore]: Personal preferences should not be in the repository.

You can use a global gitignore for this: https://help.github.com/articles/ignoring-files/

File: src/Pagerfanta/Pagerfanta.php [hunk 2]
```diff
@@ -105,6 +105,11 @@ public function getNormalizeOutOfRangePages()
         return $this->normalizeOutOfRangePages;
     }
 
+    /**
```
Comment (pablodip) [inline src/Pagerfanta/Pagerfanta.php]: I'm -1 to use PHPDoc unless it helps for something.

For me documenting is helpful to know things you can't guess.

Documenting private functions doesn't help much.
Documenting variable names which are some lines below doesn't help much, overall with good variable names which already indicate what they have.
Documenting "mixed" returns don't help much.

Comment (pablodip) [inline src/Pagerfanta/Pagerfanta.php]: Documenting is also helpful for me to indicate the IDE things that can't be guessed.

File: src/Pagerfanta/Pagerfanta.php [hunk 3]
```diff
@@ -134,6 +139,11 @@ public function setMaxPerPage($maxPerPage)
         return $this;
     }
 
+    /**
+     * @param $maxPerPage
```
Comment (stof) [inline src/Pagerfanta/Pagerfanta.php]: all your phpdoc is invalid btw. It is missing the type between ``@param`` and the name

File: src/Pagerfanta/Pagerfanta.php [hunk 4]
```diff
@@ -105,6 +105,11 @@ public function getNormalizeOutOfRangePages()
         return $this->normalizeOutOfRangePages;
     }
 
+    /**
+     * @param $value
+     *
+     * @return mixed
```
Comment (stof) [inline src/Pagerfanta/Pagerfanta.php]: the return value is always a boolean

File: src/Pagerfanta/View/Template/DefaultTemplate.php [hunk 5]
```diff
@@ -28,11 +28,19 @@ class DefaultTemplate extends Template
...
+     */
     public function container()
     {
         return $this->option('container_template');
     }
 
+    /**
+     * @param int $page
+     *
+     * @return mixed
```
Comment (stof) [inline src/Pagerfanta/View/Template/DefaultTemplate.php]: the return value is always a string

File: src/Pagerfanta/View/Template/DefaultTemplate.php [hunk 6]
```diff
@@ -28,11 +28,19 @@ class DefaultTemplate extends Template
         'span_template'      => '<span class="%class%">%text%</span>',
     );
 
+    /**
+     * @return mixed
```
Comment (stof) [inline src/Pagerfanta/View/Template/DefaultTemplate.php]: this is a string

File: src/Pagerfanta/View/Template/DefaultTemplate.php [hunk 7]
```diff
@@ -40,6 +48,12 @@ public function page($page)
         return $this->pageWithText($page, $text);
     }
 
+    /**
+     * @param int    $page
+     * @param string $text
+     *
+     * @return mixed
```
Comment (stof) [inline src/Pagerfanta/View/Template/DefaultTemplate.php]: always a string

File: src/Pagerfanta/View/Template/Template.php [hunk 8]
```diff
@@ -26,11 +26,17 @@ public function __construct()
...
+    /**
+     * @param $routeGenerator
+     */
     public function setRouteGenerator($routeGenerator)
     {
         $this->routeGenerator = $routeGenerator;
     }
 
+    /**
+     * @param array $options
```
Comment (stof) [inline src/Pagerfanta/View/Template/Template.php]: this does not add any value compared to the method signature

File: src/Pagerfanta/View/Template/Template.php [hunk 9]
```diff
@@ -26,11 +26,17 @@ public function __construct()
         $this->initializeOptions();
     }
 
+    /**
+     * @param $routeGenerator
```
Comment (stof) [inline src/Pagerfanta/View/Template/Template.php]: this phpdoc *could* be useful if it was documenting the type. Given it does not, it is totally useless (and invalid)

**Comment by chtipepere** [comment]
First, thanks for your answers, really.
Thanks too for the global gitignore ;)

I've reverted some of the PHPDOC I've added. I also normalized Boolean/bool and integer/int typings in existing PHPDOC.

File: src/Pagerfanta/Pagerfanta.php [hunk 10]
```diff
@@ -60,9 +60,9 @@ public function getAdapter()
     /**
      * Sets whether or not allow out of range pages.
      *
-     * @param Boolean $value
+     * @param bool $value
      *
-     * @return self
+     * @return Pagerfanta
```
Comment (stof) [inline src/Pagerfanta/Pagerfanta.php]: why changing this ?

File: src/Pagerfanta/Pagerfanta.php [hunk 11]
```diff
@@ -98,13 +98,19 @@ public function setNormalizeOutOfRangePages($value)
...
-     * @return Boolean
+     * @return bool
      */
     public function getNormalizeOutOfRangePages()
     {
         return $this->normalizeOutOfRangePages;
     }
 
+    /**
+     * @param bool $value
```
Comment (stof) [inline src/Pagerfanta/Pagerfanta.php]: the input is not always a boolean (which is precisely the goal of this method

File: src/Pagerfanta/View/DefaultView.php [hunk 12]
```diff
@@ -31,11 +33,17 @@ class DefaultView implements ViewInterface
     private $startPage;
     private $endPage;
 
+    /**
+     * @param TemplateInterface $template
```
Comment (stof) [inline src/Pagerfanta/View/DefaultView.php]: this does not add any value at all compared to the signature. It only duplicates knowledge (and with a mistake as the actual type is ``TemplateInterface|null``)

File: src/Pagerfanta/View/DefaultView.php [hunk 13]
```diff
@@ -54,6 +62,9 @@ public function render(PagerfantaInterface $pagerfanta, $routeGenerator, array $
         return $this->generate();
     }
 
+    /**
+     * @param PagerfantaInterface $pagerfanta
```
Comment (stof) [inline src/Pagerfanta/View/DefaultView.php]: there is no value here either

File: src/Pagerfanta/View/DefaultView.php [hunk 14]
```diff
@@ -238,64 +280,3 @@ public function getName()
...
-.pagerfanta .current {
-    background: #ccf;
-}
-
-.pagerfanta .disabled {
-    border-color: #ccf;
-    color: #cf;
-}
-
-*/
```
Comment (stof) [inline src/Pagerfanta/View/DefaultView.php]: why removing this ?

Comment (chtipepere) [inline src/Pagerfanta/View/DefaultView.php]: Why is there some css rules in this php file?
If it's useful for the implementation, why not put it in the readme? Or just add comments above to explain why it's here?

File: src/Pagerfanta/Pagerfanta.php (hunk 10 above)
Comment (chtipepere) [inline src/Pagerfanta/Pagerfanta.php]: Question of habits. But habits can be changed;)

File: src/Pagerfanta/Pagerfanta.php (hunk 11 above)
Comment (chtipepere) [inline src/Pagerfanta/Pagerfanta.php]: Indeed.

File: src/Pagerfanta/View/DefaultView.php (hunk 12 above)
Comment (chtipepere) [inline src/Pagerfanta/View/DefaultView.php]: Good point.

File: src/Pagerfanta/View/DefaultView.php (hunk 14 above)
Comment (chtipepere) [inline src/Pagerfanta/View/DefaultView.php]: In fact, it's already in the README.

**Pull request closed without merging**